pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -206,126 +206,161 @@ def applyLogMapping(self, logMode: tuple[bool, bool]):
+                    or ``None``, default ``(200, 200, 200)``
+                    
+                    Outline pen for drawing points, or a list specifying a pen for each
+                    point.
         
-        symbolBrush :class:`QBrush`, list of :class:`QBrush`, or arguments accepted by
-                    :func:`~pyqtgraph.mkBrush`, default ``(50, 50, 150)``
+        symbolBrush :class:`QBrush`, or arguments accepted by
+                    :func:`mkBrush <pyqtgraph.mkBrush>`,
+
+                    or list of :class:`QBrush`
+                    or arguments to :func:`mkBrush <pyqtgraph.mkBrush>`
+                    
+                    default ``(50, 50, 150)``
 
-                    Brush for filling points, or a list of brushes, one per point.
+                    Brush for filling points, or a list specifying a brush for each
+                    point.
         
         symbolSize  ``int`` or ``list[int]``, default ``10``
 
-                    Diameter of the symbols, or a list of diameters.  Diameter is either
-                    in pixels or data-space coordinates depending on the value of
-                    `pxMode`.
+                    Diameter of the symbols, or array-like list of diameters. Diameter
+                    is either in pixels or data-space coordinates depending on the value
+                    of `pxMode`.
         
         pxMode      ``bool``, default ``True``
 
-                    If ``True``, the `symbolSize` represents the diameter in pixels.  If
+                    If ``True``, the `symbolSize` represents the diameter in pixels. If
                     ``False``, the `symbolSize` represents the diameter in data
                     coordinates.
         =========== ====================================================================
 
-        Line Style Keyword Arguments.
+        *Line Style Keyword Arguments*
         
         =========== ====================================================================
         Property    Description
         =========== ====================================================================
-        connect     ``{ 'all', 'pairs', 'finite', 'auto', (N,) ndarray }``, default
+        connect     ``{ 'auto', 'finite', 'all', 'pairs', (N,) ndarray }``, default
                     ``'auto'``
                     
-                    - ``'auto'`` - if dataset contains non-finite values, switch to
-                      ``'all'``, otherwise change to ``'finite'``.
-                    - ``'all'`` - connects all points.  
-                    - ``'pairs'`` - generates lines between every other point.
-                    - ``'finite'`` - creates a break when a non-finite points is
-                      encountered. 
-                    - :class:`~numpy.ndarray` - it should contain `N` elements that of 
+                    - ``'auto'`` connects all points, but creates a break where a
```

Comment:
_"connects all points, but creates a break..."_ sounds contradictory.
Looking at the implementation, "auto" semantically means "finite".

In the absence of non-finites, "all" is equivalent to "finite".
Since "auto" only uses "all" when there are no non-finites, it is equivalent to using "finite".

Under the hood, the combination of "all" + `skipFiniteCheck=True` goes through a different processing path,  but that is an implementation detail.

Reply:
I know this isn't that relevant to this PR, but the `skipFiniteCheck` was intended as a band-aid, which may no longer be relevant, in which case I would be fine with removing.

Reply:
Thank you for the comments! I'll reconsider the wording to avoid the whiplash going from "connect all" to "creates breaks".

My understanding is that the mode we want to encourage is the one currently called 'auto'. It is the same as 'finite', but if e.g. log mode calculations have *already* given us the info that there are no non-finites, then the test is skipped for free.

Two alternative proposals to make this nicer and simpler:
(A) If `skipFiniteCheck` is indeed not needed anymore (is it?), then we should simply stop advertising 'auto' mode and eventually deprecate it.
(B) If the 'auto' behavior is our best default solution, then we should rename 'auto' to 'finite'. This would restore parity between the 'connect' options in PlotCurveItem and PlotDataItem. As you correctly point out, there is no functional difference on the user side. ('auto' would remain as a hidden alias for compatibility.)

Reply:
There are 3 possible modes here:
1) connect="all" + skipFiniteCheck=True
    * purpose: data is known to be a single segment with all data finite
2) connect="all" + skipFiniteCheck=False
    * purpose: data may have non-finites, but user wants to draw it as a single segment anyway (by effectively eliminating the non-finite values)
3) connect="finite" + skipFiniteCheck=False ("finite" implies that checking cannot be skipped)
    * purpose: draw runs of finite values as separate line segments

It can be observed that (1) is a special case of (3). (ignoring side-effects like whether one runs more efficiently than the other)

In the current implementation of `PlotDataItem`, `connect="auto"` will result in either (1) or (3) only. Hence, rendering-wise, `connect="auto"` is equivalent to `"finite"`.

One possible reason for the existence of (2) is that things like fill and histogram do not behave well in the presence of non-finites. i.e. (2) exists to sanitize the data for code that can't (and/or won't be updated to) handle non-finites.

Reply:
I don't understand your second bullet point about respect-(ing) `skipFiniteCheck`. Did you mean that if no known information, then use the user supplied value of `skipFiniteCheck`?

Currently, as implemented and documented, "auto" means user supplied `skipFiniteCheck` is ignored. This makes sense and is the current value-add of "auto" over "finite".
Perhaps `skipFiniteCheck` in `PlotDataItem` should be removed? (while retaining it for `PlotCurveItem`)
It can retroactively be defined that `PlotDataItem`'s `skipFiniteCheck` argument was only ever a hint that may or may not be respected.

In fact, we could follow `ImageItem`'s model where instead of just indicating whether non-finites are present, we provide (and cache) the locations (indices) of the non-finites to `PlotCurveItem`. An empty locations would indicate that all data is finite.
In such a model, `PlotDataItem` doesn't get to say that it doesn't know if the data has any non-finites. It has to find it. After all, any time saved in `PlotDataItem` just gets kicked down the line to `PlotCurveItem`.

Reply:
You are right, there is clearly an alternative architecture that would make this a lot more logical and accessible.

Having PlotDataItem always check for and locate any non-finite values would fit well with my current understanding of how processing should be split between PlotDataItem and its curve and scatter children.

A specific proposal for that will need its own PR though. For this PR, I have tried to clean up the description for the *current* state of implementation a little, but that is where I would like to leave it for now.